examples/pytorch/adaptive_sampling/adaptive_sampling.py

```diff
@@ -0,0 +1,475 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import dgl
+import dgl.function as fn
+import numpy as np
+import scipy.sparse as ssp
+from dgl.data import citation_graph as citegrh
+import networkx as nx
+
+##load data
+data = citegrh.load_cora()
+adj = nx.adjacency_matrix(data.graph)
+# reorder
+ids_shuffle = np.arange(2708)
+np.random.shuffle(ids_shuffle)
+adj = adj[ids_shuffle, :][:, ids_shuffle]
+data.features = data.features[ids_shuffle]
+data.labels = data.labels[ids_shuffle]
+##train-test split
+train_nodes = np.arange(1208)
+test_nodes = np.arange(1708, 2708)
+train_adj = adj[train_nodes, :][:, train_nodes]
+test_adj = adj[test_nodes, :][:, test_nodes]
+trainG = dgl.DGLGraph(train_adj)
+allG = dgl.DGLGraph(adj)
+h = torch.tensor(data.features[train_nodes], dtype=torch.float64)
+test_h = torch.tensor(data.features[test_nodes], dtype=torch.float64)
+all_h = torch.tensor(data.features, dtype=torch.float64)
+train_nodes = torch.tensor(train_nodes)
+test_nodes = torch.tensor(test_nodes)
+y_train = torch.tensor(data.labels[train_nodes])
+y_test = torch.tensor(data.labels[test_nodes])
+##configuration
+n_epoch=300
+lamb = 0.5
+lr = 1e-3
+weight_decay = 5e-4
+input_size = h.shape[1]
+hidden_size = 16
+output_size = data.num_labels
+##sample size for each layer during training
+layer_sizes = [256, 256]
+batch_size = 256
+##sample size for each layer during test
+test_batch_size = 64
+avg_degree = 8
```

Comment:
Simple configurations (like integers or strings) are better provided as `argparse.ArgumentParser` so one can tune them with command line arguments.  I'm not totally objecting to the current way though.